lib/jobs/ucs-discovery.js

```diff
@@ -140,44 +140,57 @@ function UcsDiscoveryJobFactory(
                         return res.body;
                     })
                     .map(function(data) {
-                        assert.array(data.macs);
-                        assert.string(data.name);
-                        assert.string(data.path);
-
-                        var config = Object.assign({}, self.settings);
-                        obm = {
-                            config: config,
-                            service: 'ucs-obm-service'
-                        };
-                        var node = {
-                            type: 'compute',
-                            name: data.path,
-                            identifiers: data.macs,
-                            obm:[obm],
-                            relations : []
-                        };
-
-                        // Update existing node  or create a new one
-                        return waterline.nodes.needOne({identifiers:data.macs})
-                            .then(function(curNode) {
-                                return waterline.nodes.updateOne(
-                                    { id: curNode.id },
-                                    node
-                                )
-                                .then(function(data){
-                                    return data;
+                        self.createNode(data, 'compute');
+                    });
+            }else{
+                logger.warning('No rackmount servers found Found');
+                return;
+            }
+        });
+    };
+
+    /**
+     * @function createServers
+     * @description discovers all the chassis servers in the Cisco UCS
+     */
+    UcsDiscoveryJob.prototype.createServers = function (root) {
```

Comment:
After examining the whole code, the cause that leads to tool meany nesting levels is that you mixed the code that creating node and updating node relations together, just split the node creation and node update to a smaller function, this function will look much more simple: